### ISSUE TITLE
BelongsTo (HasMany) Relation Unlink and Display Issues

### ISSUE THREAD

**Issue by GinoPane** [octobercms/october#3470]
##### Reproduce steps
https://stackoverflow.com/questions/49492510/octobercms-belongsto-hasmany-relation-issues

##### October build
434

Following the description from linked Stackoverflow issue and by @LukeTowers suggestion I tried to reproduce the issue using  [test plugin](https://github.com/daftspunk/oc-test-plugin).

I slightly changed `Posts` `Status` field and got this:

![image](https://user-images.githubusercontent.com/3897579/37924072-7401a086-3139-11e8-8180-fbdb8fca8333.png)

So, the layout issue was not reproduced (probably it is connected with specific styles on RainLab/BlogPost page), but the unlinking issue is still there. Here's the demo video: https://youtu.be/Zz9oOCf3UN4

This is the only change that I made to oc-test-pugin in plugins/october/test/models/post/fields.yaml:

from 

```
status:
        label: Status
        type: partial
        path: $/october/test/controllers/posts/_status.htm
        span: auto
        commentAbove: Generic model filtered by relation conditions.
        context: [create, update, relation]
```

to

```
status:
        label: Status
        type: relation
        span: auto
        commentAbove: Generic model filtered by relation conditions.
        context: [create, update, relation]
```

**Comment by GinoPane** [comment]
Moreover, "unlink" for deferred relation throws an error like the system tries to save the Post.

**Comment by GinoPane** [comment]
Unlink issue was fixed within #3476: [demo video](https://youtu.be/G7LuNH2HI3M)

**Comment by bennothommo** [comment]
Fixed by #4741.

**Comment by LukeTowers** [comment]
@bennothommo reminder about tagging issues fixed by a build with that build's milestone
